### DISCUSSION TITLE
Query and Emphasise spatial containments

### DISCUSSION THREAD

**Discussion by yibodong2001** [iTwin/community#12] [Q&A] [answered]
Hi, I am new to iTwin and I'm interested in understanding how to implement the functionality that allows querying a spatial containment (such as a restroom) and highlighting it in the iModel. I came across a video on YouTube (https://youtu.be/xqi3-sNdlzI?list=PL_HBHDZGo5EH5LI1GOGoAx0A4dNSMysqr&t=1318）that introduces this feature, but I'm unsure about how to actually implement it. Can you provide guidance or instructions on how to achieve this?

**Comment by navneetk897** [comment, marked as answer]
Hi yibodong2001,

To implement this feature you first need to query the ```iModel``` to get the properties of element such as it's ```ECInstanceID```.
You can learn about writing ECSQL queries from here https://www.itwinjs.org/learning/ecsqltutorial/
 
After having the ```ECInstanceIDs``` of the element you can highlight it by adding it in the iModelConnection's ```SelectionSet``` https://www.itwinjs.org/reference/core-frontend/selectionset/selectionset/ adding element into this set highlight the element.
This perhaps only work if ```wantSyncWithSelectionSet``` property set to true in ```HiliteSet```.

Note: You could also directly add the ```ECInstanceIDs``` of the element into the iModelConnection's  ```HiliteSet``` https://www.itwinjs.org/reference/core-frontend/selectionset/hiliteset/ but here highlighted element would not be in sync with ```SelectionSet```.

You can even define how an element should be highlighted by changing ```hilite``` settings  in viewport https://www.itwinjs.org/reference/core-common/rendering/hilite/hilite.settings/.

**Reply by yibodong2001** [reply]
Thanks so much, it helps a lot.
